include/mbgl/renderer/layer_group.hpp

```diff
@@ -0,0 +1,152 @@
+#pragma once
+
+#include <mbgl/renderer/render_pass.hpp>
+#include <mbgl/tile/tile_id.hpp>
+#include <mbgl/util/identity.hpp>
+
+#include <functional>
+#include <memory>
+#include <string>
+#include <vector>
+
+namespace mbgl {
+class LayerGroupBase;
+class PaintParameters;
+class RenderOrchestrator;
+class TileLayerGroup;
+
+using LayerGroupBasePtr = std::shared_ptr<LayerGroupBase>;
+
+namespace gfx {
+class Context;
+class Drawable;
+class UploadPass;
+
+using DrawablePtr = std::shared_ptr<Drawable>;
+using UniqueDrawable = std::unique_ptr<Drawable>;
+} // namespace gfx
+
+class LayerTweaker;
+using LayerTweakerPtr = std::shared_ptr<LayerTweaker>;
+
+/**
+    A layer-like group of drawables, not a group of layers.
```

Comment:
Should it have a different name maybe?

Reply:
We settled on this after evaluating the alternatives.  It's not perfect, but it could be worse.